include/simdjson/generic/ondemand/value_iterator-inl.h

```diff
@@ -511,7 +511,26 @@ simdjson_warn_unused simdjson_inline simdjson_result<bool> value_iterator::parse
 }
 
 simdjson_warn_unused simdjson_inline simdjson_result<std::string_view> value_iterator::get_string(bool allow_replacement) noexcept {
-  return get_raw_json_string().unescape(json_iter(), allow_replacement);
+  simdjson_result<raw_json_string> raw_json_string_result = get_raw_json_string();
+  if (raw_json_string_result.error()) {
+    return raw_json_string_result.error();
+  }
+
+  const char* start = raw_json_string_result.value_unsafe().raw();
+  const char* json = start;
+
+  for (; *json != '\0'; ++json) {
```

Comment:
You don't think you need `*json != '\0'`... and, in fact, it is likely incorrect. The JSON spec does not forbig null characters inside the string as far as I can tell, and it does not enforce that the string be null terminated.